org.eclipse.xtext/src/org/eclipse/xtext/resource/containers/StateBasedContainer.java

```diff
@@ -38,26 +35,9 @@
 	private final IContainerState state;
 
 	public StateBasedContainer(IResourceDescriptions descriptions, IContainerState state) {
-		super(descriptions);
+		super(new StateBasedResourceDescriptions(descriptions, state));
```

Comment:
Wrapping the given instance of `IResourceDescriptions` may cause trouble with implementors that check for a specific instance type (as you mentioned in a no longer available comment). Do you have another idea? Maybe using `descriptions.getAllResourceDescriptions().filter(..)` and working with that result from within this impl? Or we could add a default method to the interface `IResourceDescriptions` that allows to implement an optimized query iff the result is supposed to be filtered by URI.